src/components/LeftSidebar/LeftSidebar.vue

```diff
@@ -386,21 +407,59 @@ export default {
 			}
 		},
 
+		token() {
+			return this.$store.getters.getToken()
+		},
+
+		getFilterCaption() {
+			return this.isFiltered === 'mentions'
+				? t('spreed', 'Unread mentions')
+				: t('spreed', 'Unread messages')
+		},
+
+		messageDisplay() {
+			return this.isFiltered === 'mentions'
+				? t('spreed', 'You have no unread mentions in your inbox')
+				: t('spreed', 'You have no unread messages in your inbox')
+		},
+
 		filteredConversationsList() {
 			if (this.isFocused) {
 				return this.conversationsList
 			}
 
+			// applying filters
+
 			if (this.isFiltered === 'unread') {
-				return this.conversationsList.filter(conversation => conversation.unreadMessages > 0
-				|| conversation.hasCall || conversation.token === this.$store.getters.getToken())
+				const filteredConversations = this.conversationsList.filter(function(conversation) {
+					return this.hasUnreadMessages(conversation)
+							|| conversation.hasCall
+							|| conversation.token === this.token
+				}.bind(this))
+
+				// return empty if it only includes the current conversation without any flags
+				const includesCurrentConversationOnly = filteredConversations.length === 1
+					&& filteredConversations.filter(function(conversation) {
+						return !this.hasUnreadMessages(conversation) && conversation.token === this.token
+					}.bind(this)).length
+
+				return includesCurrentConversationOnly ? [] : filteredConversations
 			}
 
 			if (this.isFiltered === 'mentions') {
-				return this.conversationsList.filter(conversation => conversation.unreadMention
-				|| conversation.hasCall
-				|| (conversation.unreadMessages > 0 && (conversation.type === CONVERSATION.TYPE.ONE_TO_ONE || conversation.type === CONVERSATION.TYPE.ONE_TO_ONE_FORMER))
-				|| conversation.token === this.$store.getters.getToken())
+				const filteredConversations = this.conversationsList.filter(function(conversation) {
```

Comment:
Looks like a duplicated logic. Maybe move everything to the method, with this.isFiltered as filtration argument?